workers/tests/unittests/test_config.py

```diff
@@ -0,0 +1,102 @@
+import inspect
+import os
+import tempfile
+import unittest
+from unittest import mock
+
+from ohsome_quality_analyst.config import config
+
+
+class TestConfig(unittest.TestCase):
+    @mock.patch.dict("os.environ", {}, clear=True)
+    def test_get_config_path_empty_env(self):
+        self.assertEqual(
+            config.get_config_path(),
+            os.path.join(
+                os.path.dirname(
+                    inspect.getfile(config),
+                ),
+                "config.yaml",
+            ),
+        )
+
+    @mock.patch.dict(
+        "os.environ",
+        {"OQT_CONFIG": os.path.join(tempfile.gettempdir(), "config.yaml")},
+        clear=True,
+    )
+    def test_get_config_path_set_env(self):
+        self.assertEqual(
+            config.get_config_path(),
+            os.path.join(tempfile.gettempdir(), "config.yaml"),
+        )
+
+    def test_load_config_from_file(self):
+        path = config.get_config_path()
+        cfg = config.load_config_from_file(path)
+        self.assertIsInstance(cfg, dict)
+        self.assertTrue(cfg)
+
+    @mock.patch.dict("os.environ", {}, clear=True)
+    def test_load_config_from_env_empty(self):
+        cfg = config.load_config_from_env()
+        self.assertEqual(cfg, {})
+
+    @mock.patch.dict(
+        "os.environ",
+        {"OQT_GEOM_SIZE_LIMIT": "200", "POSTGRES_HOST": "foo"},
+        clear=True,
+    )
+    def test_load_config_from_env_set(self):
+        cfg = config.load_config_from_env()
+        self.assertDictEqual(
+            cfg,
+            {"geom_size_limit": "200", "postgres_host": "foo"},
+        )
+
+    @mock.patch.dict("os.environ", {}, clear=True)
+    def test_get_config(self):
+        cfg = config.get_config()
+        self.assertIsInstance(cfg, dict)
+        self.assertTrue(cfg)
+
+    def test_check_missing_conf_keys_compete(self):
+        cfg = {
+            "postgres_host": None,
+            "postgres_port": None,
+            "postgres_db": None,
+            "postgres_user": None,
+            "postgres_password": None,
+            "data_dir": None,
+            "geom_size_limit": None,
+            "log_level": None,
+            "ohsome_api": None,
+            "test_db": None,
+            "user_agent": None,
+            "datasets": None,
+        }
+        cfg = config.check_missing_conf_keys(cfg)
+        self.assertEqual(cfg, cfg)
+
+    def test_check_missing_conf_keys_missing(self):
+        cfg = {"postgres_host": None}
+        with self.assertRaises(KeyError):
+            config.check_missing_conf_keys(cfg)
+
+    def test_get_data_dir_unset_env(self):
+        data_dir = config.get_default_data_dir()
+        expected = os.path.join(
+            os.path.dirname(
+                os.path.abspath(__file__),
+            ),
+            "..",
+            "..",
+            "..",
+            "data",
+        )
+        self.assertTrue(os.path.samefile(data_dir, expected))
```

Comment:
This tests the same implementation in the test and the method. I would suggest to use a string instead.

Reply:
But then it is only valid for linux systems is it not.